### PR TITLE
Blog post on dashboards

### PR DESCRIPTION
**Pull request by robmarcer**
## Description

Created a blog post comparing the most popular choices for building NR dashboards in 2023.

### CODE REVIEW

File: src/blog/2023/03/comparinig-dashboards.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+Credit to the developers for adding in this helpful text but I suspect most users will start off by installing FlexDash then later discover that was not the correct way to proceed. It would be great if the custom node which needs to be installed was the one called 'FlexDash' in my opinion.
+
+This problem is compounded by their being no help file at all for 'FlexDash' showing up on the Node-RED web site. I suspect that may be a deliberate attempt to get users to reconsider if they are looking at the right custom node but it was still a confusing start for me and I suspect many users will have a similar experience.
```
Comment (Steve-Mcl) [inline src/blog/2023/03/comparinig-dashboards.md:63]:
```suggestion
This problem is compounded by their being no help file at all for 'FlexDash' showing up on the Node-RED web site. That may well be a deliberate attempt to help users get the right custom node installed, but it was still a confusing start for me and I suspect other users will have a similar experience.
```

File: src/blog/2023/03/comparinig-dashboards.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+When setting up FlexDash, one thing that wasn't immediately obvious was that I needed to restart Node-RED before the custom node showed in the palette. This step is [covered in the docs](https://flexdash.github.io/docs/quick-start/#installing-flexdash-in-node-red) but I suspect a lot of users will get stuck working out why the palette manager says the custom node is installed but nothing new has been added to the palette.
+
+I believe a few improvements to the install process could make FlexDash a much more popular custom node.
```
Comment (Steve-Mcl) [inline src/blog/2023/03/comparinig-dashboards.md:67]: May wish to add something about a resolution being actively discussed here: https://github.com/node-red/node-red/issues/569#issuecomment-1460556901

File: src/blog/2023/03/comparinig-dashboards.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+Dashboard is now in a a maintenance only state. In the words of the project lead, *'Angular 1 (the framework used to build Dashboard) is now unsupported and it's just a matter of time before there is a serious security hole raised against it, for which there will be no fix. Of course we don't use all the features of it so we may be lucky that an exploit doesn't necessarily expose us directly but it will compromise any audits people may wish to do'*.
+
+In practice, this means that sooner or later using Dashboard might become a significant security risk.
+
+## What are the future development plans?
+
+### uibuilder - joint firstplace
+
+During the writing of this article, uibuilder released a new version with some significant new features. In the words of the project lead when talking about version 1.6, '*It feels like uibuilder really is growing up. No more apologising for not being a direct Node-RED Dashboard replacement, uibuilder has its own path*.
```
Comment (Steve-Mcl) [inline src/blog/2023/03/comparinig-dashboards.md:149]:
```suggestion
During the writing of this article, uibuilder released a new version with some significant new features. In the words of the project lead when talking about version 6.1.0, '*It feels like uibuilder really is growing up. No more apologising for not being a direct Node-RED Dashboard replacement, uibuilder has its own path*.
```

File: src/blog/2023/03/comparinig-dashboards.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+If you already use Dashboard, in a non-commercial setting you should probably continue to do so, you might find that its development slows down to a near stall due to the underlying framework now being abandoned but for at least as of right now it's a great solution to build your Node-RED dashboards in.
+
+FlexDash is probably the best low-code solution for building dashboards in Node-RED. If you don't get blocked by the confusing install process I believe it's the one to pick up at the time of writing due to it's ongoing support and low-code interface.
+
+uibuilder is currently not really a low-code solution to creating dashboards but it's quickly moving in that direction. It's got some amazing features and has a good chance of ending up as the most popular solution to build dashboards in Node-RED in the long term. That being said, as of time of writing unless you are a 'coder' you will probably struggle to build useful dashboards using it.
```
Comment (Steve-Mcl) [inline src/blog/2023/03/comparinig-dashboards.md:179]:
```suggestion
uibuilder is currently not what I would consider a truly low-code option for creating dashboards but it is moving in that direction. It has some great features and is extremely flexible so it has a good chance of ending up as the most popular solution to build dashboards in Node-RED in the long term. That being said, as of time of writing unless you are a 'coder' you will may struggle to build a dashboard using it.
```

Open to opinion on reqording.

File: src/blog/2023/03/comparinig-dashboards.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+- [How good is the support and documentation?](#how-good-is-the-support-and-documentation%3F)
+
+- [How 'cloud native' is the dashboard?](#how-'cloud-native'-is-the-dashboard%3F)
+
+- [How active is each project's development?](#how-active-is-each-project's-development%3F)
+
+- [What are the future development plans?](#what-are-the-future-development-plans%3F)
+
+I am assuming the user is a low-code developer. They may have limited experience with coding but they are most comfortable working in visual interfaces.
```
Comment (iskerrett) [inline src/blog/2023/03/comparinig-dashboards.md:41]:
```suggestion
I am assuming the user is a low-code developer. They may have limited experience with coding and are most comfortable working in visual interfaces.
```

File: src/blog/2023/03/comparinig-dashboards.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+
+I am also assuming that the user has been recommended to search for each of these dashboards by name. It's a little unfair to say that Dashboard is inherently better because its name is an obvious search term for 'dashboards'.
+
+So, that's the methodology, let's get on with looking at the strengths of each project.
+
+## How easy is it to install?
+
+### uibuilder - 1st place
+
+A search on [DuckDuckGo for the term 'node red ui builder'](https://duckduckgo.com/?q=node+red+ui+builder&t=h_&ia=web) returns the correct custom node. When searching for the custom node in the palette manager there is only one result, this is great as users are very likely to install what they were searching for. Once you've found the correct custom node, the installation takes just a few moments using the palette manager.
```
Comment (iskerrett) [inline src/blog/2023/03/comparinig-dashboards.md:51]: I am curious why you are using DuckDuckGO for search. A very large majority of people use Google for search?

Comment (robmarcer) [reply src/blog/2023/03/comparinig-dashboards.md:51]: DDG gives the same results to all users where as Google search is very inconsistent. I feel using DDG is a more valid way to understand current search rankings but I don't mind changing it.

File: src/blog/2023/03/comparinig-dashboards.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+
+- [How 'cloud native' is the dashboard?](#how-'cloud-native'-is-the-dashboard%3F)
+
+- [How active is each project's development?](#how-active-is-each-project's-development%3F)
+
+- [What are the future development plans?](#what-are-the-future-development-plans%3F)
+
+I am assuming the user is a low-code developer. They may have limited experience with coding but they are most comfortable working in visual interfaces.
+
+I am also assuming that the user has been recommended to search for each of these dashboards by name. It's a little unfair to say that Dashboard is inherently better because its name is an obvious search term for 'dashboards'.
```
[the remark this answers is not in the document]
Comment (robmarcer) [reply src/blog/2023/03/comparinig-dashboards.md:43]: The point being, if you don't know what you are searching for you will always end up using Dashboard is its name is synonymous with dashboards. I can remove it if you feel it's redundant?

File: src/blog/2023/03/comparinig-dashboards.md (hunk 2 above)
Comment (robmarcer) [reply src/blog/2023/03/comparinig-dashboards.md:69]: Added to both places I've mentioned the install problem, thanks!

**Review by iskerrett — APPROVED** [review]

**Pull request merged**